controller/src/main/java/io/pravega/controller/task/Stream/StreamTransactionMetadataTasks.java

```diff
@@ -53,38 +51,54 @@
  * Instead, a new overloaded method may be created with the same task annotation name but a new version.
  */
 @Slf4j
-public class StreamTransactionMetadataTasks extends TaskBase {
+public class StreamTransactionMetadataTasks implements AutoCloseable {
 
     protected EventStreamWriter<CommitEvent> commitEventEventStreamWriter;
     protected EventStreamWriter<AbortEvent> abortEventEventStreamWriter;
     protected String commitStreamName;
```

Comment:
We could initialize them in constructor, but we preferred collocating initialization of related properties. Hence we initialize all stream related properties including stream writers and stream names in the initializeStreamWriters method.